clang/include/clang/AST/DeclBase.h

```diff
@@ -306,6 +306,11 @@ class alignas(8) Decl {
   /// are regarded as "referenced" but not "used".
   unsigned Referenced : 1;
 
+  /// Whether the last reference to this declaration happened in Coroutine
+  /// Parameter moves. Otherwise the reference caused by such moves would
+  /// prevent a warning against unused parameters in all coroutines.
+  unsigned LastReferenceInCoroutineParamMoves : 1;
```

Comment:
Let's try to not adding new bits to `Decl`. It will add burdens to other Declarations which is not related to this. Also the patch doesn't try to serialize/deserialize the bit. So it should be incorrect. Even if we tried to do that, I think it is still a cost than an improvement.